src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/SearchParameterExpressionParser.cs

```diff
@@ -199,19 +199,38 @@ public SearchParameterExpressionParser(IReferenceSearchValueParser referenceSear
                 }
 
                 // This is a multiple value expression.
-                Expression[] expressions = parts.Select(part =>
+                if (modifier == SearchModifierCode.Not)
                 {
-                    ISearchValue searchValue = parser(part);
+                    Expression[] expressions = parts.Select(part =>
```

Comment:
I might suggest consolidating the two very similar branches to something like
```
var orExpression = Expression.Or(
    parts.Select(part =>
    {
        ISearchValue searchValue = parser(part);

        return helper.Build(
            searchParameter.Name,
            modifier == SearchModifierCode.Not ? null : modifier,
            comparator,
            componentIndex,
            searchValue);
    }).ToArray());

return modifier == SearchModifierCode.Not ? Expression.Not(orExpression) : orExpression;
```

Reply:
I did consider this. But decided in favor of 2 distinct branches.
My reasoning is:
> We need to do 2 things using the same if statement, which I think is easier to read if the branches are split.
> Also, performance consideration, running IF N+1 times instead of 1 (which doesn't really matter in this case).

Reply:
Ok that's fair.